Neroc/markdigui

nerocui opened the pull request.

nerocui committed:
initial commit

nerocui committed:
fixed dependencies

niels9001 commented on components/MarkdigUI/src/CommunityToolkit.WinUI.Controls.MarkdigUI.csproj:
```diff
@@ -0,0 +1,29 @@
+  <!-- Sets this up as a toolkit component's source project -->
+  <Import Project="$(ToolingDirectory)\ToolkitComponent.SourceProject.props" />
+  <ItemGroup>
+    <PackageReference Include="ColorCode.Core" Version="2.0.14" />
```
Remove package references here

niels9001 commented on components/MarkdigUI/src/Dependencies.props:
```diff
@@ -0,0 +1,31 @@
+    
+    <!-- WinUI 3 / Uno -->
+    <ItemGroup Condition="'$(IsUno)' == 'true' AND '$(WinUIMajorVersion)' == '3'">
+        <!-- <PackageReference Include="Uno.CommunityToolkit.WinUI.UI.Controls.Primitives" Version="7.1.100-dev.15.g12261e2626"/> -->
```
`<PackageReference Include="Uno.ColorCode.WinUI" Version="2.1.0-uno.36" />`

niels9001 commented on components/MarkdigUI/src/Dependencies.props:
```diff
@@ -0,0 +1,31 @@
+
+    <!-- WinUI 2 / Uno -->
+    <ItemGroup Condition="'$(IsUno)' == 'true' AND '$(WinUIMajorVersion)' == '2'">
+        <!-- <PackageReference Include="Uno.Microsoft.Toolkit.Uwp.UI.Controls.Primitives" Version="7.1.11"/> -->
```
`<PackageReference Include="Uno.ColorCode.UWP" Version="2.1.0-uno.36" />`

niels9001 commented on components/MarkdigUI/src/Dependencies.props:
```diff
@@ -0,0 +1,31 @@
+<Project>
+    <!-- WinUI 2 / UWP -->
+    <ItemGroup Condition="'$(IsUwp)' == 'true'">
+        <!-- <PackageReference Include="Microsoft.Toolkit.Uwp.UI.Controls.Primitives" Version="7.1.2"/> -->
```
`<PackageReference Include="ColorCode.UWP" Version="2.0.14" />`

niels9001 commented on components/MarkdigUI/src/Dependencies.props:
```diff
@@ -0,0 +1,31 @@
+
+    <!-- WinUI 3 / WinAppSdk -->
+    <ItemGroup Condition="'$(IsWinAppSdk)' == 'true'">
+        <!-- <PackageReference Include="CommunityToolkit.WinUI.UI.Controls.Primitives" Version="7.1.2"/> -->
```
`<PackageReference Include="ColorCode.WinUI" Version="2.0.14" />`

niels9001 commented on components/MarkdigUI/src/MultiTarget.props:
```diff
@@ -0,0 +1,9 @@
+          MultiTarget is a custom property that indicates which target a project is designed to be built for / run on.
+          Used to create project references, generate solution files, enable/disable TargetFrameworks, and build nuget packages.
+        -->
+        <MultiTarget>uwp;wasdk;wpf;wasm;linuxgtk;macos;ios;android;</MultiTarget>
```
Are you planning to support Uno.WASM as well? If not, you can remove everything after `uwp;wasdk;`

Might make it easier for initial testing as well :)

nerocui committed:
added dependencies in implicit global usings

nerocui committed:
fixed some references

niels9001 committed:
Remove redundant sample references

niels9001 committed:
Remove Uno target heads

niels9001 committed:
Adding file heads and missing namespaces for UWP

niels9001 committed:
Surpressing null reference errors

nerocui committed:
fixed crash

niels9001 committed:
Fix markdown not showing

nerocui committed:
polished the experiment

nerocui committed:
changed name to MarkdownTextBlock

nerocui committed:
fixed xaml theme

nerocui committed:
fixed icon issue

nerocui committed:
updated description

nerocui committed:
added back the sample

nerocui committed:
added example for task list

nerocui committed:
making it StatusAndInfo instead of Layout

nerocui committed:
made requested changes

changed renderer name, made text separate prop, pipeline is now cached between renders

nerocui committed:
bind to textbox instead of a property

nerocui committed:
run styler

nerocui committed:
addressed perf issue

nerocui committed:
fixed code style

nerocui committed:
added theme object, abstracted some styling to the theme obj

nerocui committed:
clean up

nerocui committed:
fixed fontweight using statement

nerocui committed:
removing unused code

nerocui committed:
handled switch statement exhaustively.

nerocui committed:
Merge branch 'CommunityToolkit:main' into neroc/markdigui

nerocui committed:
added header

nerocui committed:
fixed test failures

added label and new icon

niels9001 committed:
Merge branch 'main' into neroc/markdigui

nerocui committed:
Update Dependencies.props

nerocui committed:
Revert "Update Dependencies.props"

This reverts commit 80c8cf2ec319d48c9b9c0680147c970f5b15e5da.

Arlodotexe committed:
Update tooling to latest main

Arlodotexe committed:
Merge pull request #524 from CommunityToolkit/fix/ci/pwsh/7.4.0

Fix CI failure after update to PowerShell 7.4.0

niels9001 committed:
Merge branch 'main' into neroc/markdigui

nerocui committed:
commented out colorcode for now

There will be bug in code block, we will fix it in the next PR.

nerocui committed:
Merge branch 'neroc/markdigui' of https://github.com/nerocui/Labs-Windows into neroc/markdigui

Arlodotexe committed:
Update components/MarkdownTextBlock/src/CommunityToolkit.WinUI.Controls.MarkdownTextBlock.csproj

michael-hawker committed:
Merge pull request #480 from nerocui/neroc/markdigui

[Experiment] Markdig based MarkdownTextBlock
